### PR TITLE
 [Reland] [11/N] Enable clang-tidy warnings on c10/util/*.h

### PR DESCRIPTION
**Pull request by cyyever**
Reland of #116353 with C++ diagnostic macros restored.

cc @izaitsevfb

### CODE REVIEW

**Comment by cyyever** [comment]
@izaitsevfb It should fix the inner builds.

**Review by albanD — APPROVED** [review]
Thanks for the update!

File: c10/util/Half.h [hunk 1]
```diff
@@ -18,10 +18,8 @@
 
 #if defined(__cplusplus) && (__cplusplus >= 201103L)
 #include <cmath>
-#include <cstdint>
```
Comment (Skylion007) [inline c10/util/Half.h:21]: I think removing these did cause a compile error that patched in a different PR.

**Pull request closed without merging**